src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -145,14 +145,34 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
     val spark = df.sparkSession
     val dfColumnNames = df.schema.fieldNames
     val indexedColumns = indexConfig.indexedColumns
-    val includedColumns = indexConfig.includedColumns
+    val includeColumns = indexConfig.includedColumns.include
+    val excludeColumns = indexConfig.includedColumns.exclude
     val resolvedIndexedColumns = ResolverUtils.resolve(spark, indexedColumns, dfColumnNames)
-    val resolvedIncludedColumns = ResolverUtils.resolve(spark, includedColumns, dfColumnNames)
+    val resolvedIncludeColumns = ResolverUtils.resolve(spark, includeColumns, dfColumnNames)
+    val resolvedExcludeColumns = ResolverUtils.resolve(spark, excludeColumns, dfColumnNames)
+
+    (resolvedIndexedColumns, resolvedIncludeColumns, resolvedExcludeColumns) match {
+      case (Some(indexed), Some(include), Some(exclude)) =>
+        if (include.intersect(exclude).nonEmpty) {
+          throw HyperspaceException(
+            "IncludedColumns include and exclude columns can not overlap.")
+        }
+
+        // If IndexConfig has both 'include' and 'exclude' columns, we pick the columns
+        // listed in indexConfig.include as index's included columns and ignore `exclude`.
+        if (includeColumns.nonEmpty) {
+          return (indexed, include)
+        } else if (excludeColumns.nonEmpty) {
+          val columns = dfColumnNames.filterNot(c => indexed.contains(c) || exclude.contains(c))
+          return (indexed, columns)
+        }
+
+        // There is no 'include' or 'exclude' column lists in indexConfig. So index will be created
+        // with only indexed columns and without any included column.
+        (indexed, Seq[String]())
```

Comment:
In this case, can we include all columns except indexed columns? I think that is a more expected behavior.

#229 

> The user does not specify included columns - take all the columns (minus indexed columns) as included columns.

Reply:
This is added to be as close as possible to today's behavior: when a user wants to create an index with only indexed columns, they simply set empty included columns seq. If we make the change you suggest, then in-order to create such an index, the user has to explicitly add each and every columns to the `exclude` list.

Reply:
Ok then.. any idea for including all columns without specifying them?
I think it would be good to have that option and it's one requirement of that issue.

Reply:
An index is typically created on a subset of columns and not the whole data. Given that picking each an every column in source data should be a rare case, one can do that by listing all columns (minus indexed ones) in the include list of IncludedColumnsConfig.

Reply:
In the reporting issue(#229) and even in the pr description, there's an example picking all columns.

>Exclude no column from schema and pick all columns as included columns. So index will have {"C1", "C2", "C3", "C4"} as its included columns:
val cols = IncludedColumns(exclude = Seq())

Hyperspace indexes can be used to remove shuffle or improve filter query performance, so I think it's a bit different from  traditional indexes.

Reply:
My prev comment does not say that an index can't/shouldn't be created on all columns, and obviously yes I know one can use an index to remove shuffles. 
Here we really talk about two cases:
1. Creating an index with only indexed columns.
2. Creating an index on all columns of source data.

We also have two syntax:
1. Listing no include/exclude column. (Simpler syntax)
2. Listing each and every column as include OR exclude.

We are trying to assign these two syntax to above cases. (Do you agree? :))
I am saying that, first case happens more frequent than second and by using an `IncludedColumns` with no include/exclude columns for the first case, we are:
1. Using the simpler syntax for a more frequent case, and
2. Keep this syntax change (switching from `Seq[String]()` to `IncludedColumns`) closer to what we have today. (When user wants to create an index with only indexed columns, he passes an empty sequence).

If we go with your suggestion, then the simpler syntax will be used for the second case, which is a more rare case to happen and that is why I am not in favor of it. @sezruby Let me know how you think.

Reply:
Yea I agree with your point (empty included & excluded columns => only indexed columns) but how could we support both (included all or nothing) cases? Because specifying all columns is also not a good API in case there are many columns in source data.

Reply:
I agree with you, I am in favor of avoiding necessity of listing all columns.
As both `include` and `exclude` should not be set together, one option could be using:
1. One Seq[String] () to list column(s) names, and
2.  Using a `mode` to specify `include` or `exclude`.

(This is somehow related to the [other comment's discussion](https://github.com/microsoft/hyperspace/pull/309#discussion_r557067683))

Reply:
Yep this looks better in terms of 1) receiving 1 list 2) we can support both including all or nothing

Could you update PR? Thanks!